vulkan/vulkan_device.cc

```diff
@@ -0,0 +1,320 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/vulkan/vulkan_device.h"
+#include <limits>
+#include <vector>
+#include "flutter/vulkan/vulkan_proc_table.h"
+#include "flutter/vulkan/vulkan_surface.h"
+#include "flutter/vulkan/vulkan_utilities.h"
+#include "third_party/skia/include/gpu/vk/GrVkBackendContext.h"
+
+namespace vulkan {
+
+static uint32_t FindGraphicsQueueIndex(
+    const std::vector<VkQueueFamilyProperties>& properties) {
+  for (uint32_t i = 0, count = static_cast<uint32_t>(properties.size());
+       i < count; i++) {
+    if (properties[i].queueFlags & VK_QUEUE_GRAPHICS_BIT) {
+      return i;
+    }
+  }
+  return std::numeric_limits<uint32_t>::max();
```

Comment:
Maybe break this out into a "constexpr kVulkanInvalidIndex", used here and below.